Source/ARTChannel.m

```diff
@@ -73,6 +74,17 @@ - (void)publish:(art_nullable NSString *)name data:(art_nullable id)data extras:
         if (callback) callback([ARTErrorInfo createFromNSError:error]);
         return;
     }
+    
+    // Checked after encoding, so that the client can receive callback with encoding errors
+    if ([self exceedMaxSize:@[message]]) {
+        ARTErrorInfo *sizeError = [ARTErrorInfo createWithCode:40009
```

Comment:
Is this a common error between all libs?

Reply:
I think we should add those error codes in an enum or at least have it in a global&static var, even though, I know there are a bunch of hard coded codes in the source. Maybe raising an issue is appropriated. What do you think?

Reply:
Make sense. Issue here: https://github.com/ably/ably-ios/issues/760